gremlin-dotnet/example/Example.cs

```diff
@@ -0,0 +1,131 @@
+    // Iterate() does not return any data and is used to just generate side-effects (i.e. write data to the database)
+    g.V(v1).AddE("knows").To(v2).Property("weight", 0.75).Iterate();
+    g.V(v1).AddE("knows").To(v3).Property("weight", 0.75).Iterate();
+
+    // Retrieve the data from the "marko" vertex
+    var marko = g.V().Has("person","name","marko").Values<string>("name").Next();
+    Console.WriteLine("name: " + marko);
+    // Find the "marko" vertex and then traverse to the people he "knows" and return their data
+    var peopleMarkoKnows = g.V().Has("person","name","marko").Out("knows").Values<string>("name").ToList();
+    foreach (var person in peopleMarkoKnows)
+    {
+        Console.WriteLine("marko knows " + person);
+    }
+}
+
+static void ModernTraversalExample()
+{
+    var server = new GremlinServer("localhost", 8182);
+    using var remoteConnection = new DriverRemoteConnection(new GremlinClient(server), "g");
+    var g = Traversal().WithRemote(remoteConnection);
+
+    /*
+    This example requires the Modern toy graph to be preloaded upon launching the Gremlin server.
+    For details, see https://tinkerpop.apache.org/docs/current/reference/#gremlin-server-docker-image and use
+    conf/gremlin-server-modern.yaml.
+    */
+    var e1 = g.V(1).BothE().ToList(); // (1)
+    var e2 = g.V(1).BothE().Where(OtherV().HasId(2)).ToList(); // (2)
+    var v1 = g.V(1).Next();
+    var v2 = g.V(2).Next();
+    var e3 = g.V(v1).BothE().Where(OtherV().Is(v2)).ToList(); // (3)
+    var e4 = g.V(v1).OutE().Where(InV().Is(v2)).ToList(); // (4)
+    var e5 = g.V(1).OutE().Where(InV().Has(T.Id, Within(2, 3))).ToList(); // (5)
+    var e6 = g.V(1).Out().Where(__.In().HasId(6)).ToList(); // (6)
+
+    Console.WriteLine(string.Join(", ", e1));
+    Console.WriteLine(string.Join(", ", e2));
+    Console.WriteLine(string.Join(", ", e3));
+    Console.WriteLine(string.Join(", ", e4));
+    Console.WriteLine(string.Join(", ", e5));
+    Console.WriteLine(string.Join(", ", e6));
+
+    /*
+    1. There are three edges from the vertex with the identifier of "1".
+    2. Filter those three edges using the where()-step using the identifier of the vertex returned by otherV() to
+       ensure it matches on the vertex of concern, which is the one with an identifier of "2".
+    3. Note that the same traversal will work if there are actual Vertex instances rather than just vertex
+       identifiers.
+    4. The vertex with identifier "1" has all outgoing edges, so it would also be acceptable to use the directional
+       steps of outE() and inV() since the schema allows it.
+    5. There is also no problem with filtering the terminating side of the traversal on multiple vertices, in this
+       case, vertices with identifiers "2" and "3".
+    6. There’s no reason why the same pattern of exclusion used for edges with where() can’t work for a vertex

```

Comment:
Prefer to use C# style step syntax
```suggestion
    2. Filter those three edges using the Where()-step using the identifier of the vertex returned by OtherV() to
       ensure it matches on the vertex of concern, which is the one with an identifier of "2".
    3. Note that the same traversal will work if there are actual Vertex instances rather than just vertex
       identifiers.
    4. The vertex with identifier "1" has all outgoing edges, so it would also be acceptable to use the directional
       steps of OutE() and InV() since the schema allows it.
    5. There is also no problem with filtering the terminating side of the traversal on multiple vertices, in this
       case, vertices with identifiers "2" and "3".
    6. There’s no reason why the same pattern of exclusion used for edges with Where() can’t work for a vertex

```